templates/content_based.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Content Based Movies</title>
+    <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='styles.css') }}" />
+</head>
+<body>
+     <div class="navigation">
+        <a href="{{ url_for('popular') }}">Popular</a>
+        <a href="{{ url_for('content_based_html') }}">Content Based</a>
+        <a href="{{ url_for('collaborative_html') }}">Collaborative</a>
+    </div>
+    <div class="search-results">
+        <h1>Content Based</h1>
+    </div>
+    <div class="movie-cards" id="movie-cards">
+        {% for movie in movies %}
```

Comment:
The movies variable is used in the for loop, but it doesn't seem to be defined in the provided code. Make sure that the movies variable is properly defined before rendering this template.